src/sentry/models/grouphistory.py

```diff
@@ -191,7 +191,13 @@ def record_group_history_from_activity_type(
     Writes a `GroupHistory` row for an activity type if there's a relevant `GroupHistoryStatus` that
     maps to it
     """
-    status = activity_type_to_history_status(activity_type)
+    status = {
+        Activity.SET_IGNORED: GroupHistoryStatus.IGNORED,
+        Activity.SET_RESOLVED: GroupHistoryStatus.RESOLVED,
+        Activity.SET_RESOLVED_IN_COMMIT: GroupHistoryStatus.SET_RESOLVED_IN_COMMIT,
+        Activity.SET_RESOLVED_IN_RELEASE: GroupHistoryStatus.SET_RESOLVED_IN_RELEASE,
+        Activity.SET_UNRESOLVED: GroupHistoryStatus.UNRESOLVED,
+    }.get(activity_type, None)
```

Comment:
Nit: It's slightly better to declare this as a module variable so that it isn't defined every time we call this method.

Reply:
Hmmm, interesting. I tried looking and I don't think python optimizes literals into constant references like what java does.

Reply:
Yeah, unfortunately it doesn't tend to do that kind of optimization.